crypto/openssl/crypto/aes/asm/aesv8-armx.pl

```diff
@@ -2346,10 +2356,15 @@ ()
 .Lxts_enc_big_size:
 ___
 $code.=<<___	if ($flavour =~ /64/);
+#ifdef __CHERI_PURE_CAPABILITY__
+	str	$tmpinp,[csp,#-80]!
+	str	$constnumx,[csp,#64]
+#else
 	stp	$constnumx,$tmpinp,[sp,#-64]!
-	stp	$tailcnt,$midnumx,[sp,#48]
-	stp	$ivd10,$ivd20,[sp,#32]
-	stp	$ivd30,$ivd40,[sp,#16]
```

Comment:
This chunk here (and it's complement) are not quite right, we need to save/restore cap registers always, not sometimes X since we don't know what the old register might be.  I've reworked this locally and will retest.